### DISCUSSION TITLE
When ExportUnsampledSpans is set to true,  will the spans exported to collector using OTLP gRPC contain trace flags?

### DISCUSSION THREAD

**Discussion by wdengw** [open-telemetry/opentelemetry-java#6741] [Q&A]
[Issue #6057](https://github.com/open-telemetry/opentelemetry-java/pull/6057) implemented ExportUnsampledSpans flag,  setting it to true will export all spans to collector no matter a span is sampled or not.  This is great! we are looking for this feature.  A question regarding this feature is that,  will the span data exported contains TraceFlag information for collector to find out whether a received span is sampled or not?  @HaloFour and @jack-berg

**Comment by trask** [comment]
yes, they should be available via https://github.com/open-telemetry/opentelemetry-proto/blob/a8f08fc49d60538f97ffabcc7feac92f832976dd/opentelemetry/proto/trace/v1/trace.proto#L112-L133

**Reply by wdengw** [reply]
@trask Thank you!  This is great! Now trace proto has a field to record the TraceFlags.  

One follow up question regarding the implementation of https://github.com/open-telemetry/opentelemetry-java/pull/6057: Should the configuration option of exportUnsampledSpans that is added to BatchSpanProcessor and SimpleSpanProcessor be also added to [AlwaysOffSampler](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/trace/src/main/java/io/opentelemetry/sdk/trace/samplers/AlwaysOffSampler.java#L28C12-L28C35) for it to return _**SamplingResult.recordOnly()**_ when exportUnsampledSpans is set to true.

**Reply by trask** [reply]
> Should the configuration option of exportUnsampledSpans that is added to BatchSpanProcessor and SimpleSpanProcessor be also added to [AlwaysOffSampler](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/trace/src/main/java/io/opentelemetry/sdk/trace/samplers/AlwaysOffSampler.java#L28C12-L28C35)

I'm not sure, since the AlwaysOffSampler is defined in the specification: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#alwaysoff

are you able to create your own sampler and use that?

**Reply by wdengw** [reply]
To make it work for now.  we created a **_RecordOnlySampler_** that returns recordOnly Sampling result.  We use it to replace  AlwaysOffSampler when we configure the TraceProvider when exportUnsampledSpans set to true.

**Reply by wdengw** [reply]
Here is an enhancement suggestion I filed sometime back: https://github.com/open-telemetry/opentelemetry-specification/issues/2986.
